web/modules/custom/rdf_entity/src/Entity/Controller/RdfListBuilder.php

```diff
@@ -4,43 +4,65 @@
 
 use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Entity\EntityListBuilder;
+use Drupal\rdf_entity\Form\RdfListBuilderFilterForm;
 
 /**
  * Provides a list controller for rdf_entity entity.
- *
- * @ingroup content_entity_example
  */
 class RdfListBuilder extends EntityListBuilder {
+
+  /**
+   * The pager size.
+   *
+   * @var int
+   */
   protected $limit = 20;
 
   /**
    * {@inheritdoc}
    */
-  public function load() {
-    /** @var \Drupal\rdf_entity\Entity\RdfEntitySparqlStorage $rdf_storage */
+  protected function getEntityIds() {
+    $request = \Drupal::request();
     $rdf_storage = $this->getStorage();
+    /** @var \Drupal\Core\Entity\EntityTypeBundleInfoInterface $bundle_info */
+    $bundle_info = \Drupal::service('entity_type.bundle.info');
+    /** @var \Drupal\rdf_entity\Entity\Query\Sparql\Query $query */
+    $query = $rdf_storage->getQuery();
 
-    $query = $rdf_storage->getQuery()->condition('rid', NULL, 'IN');
     // If a graph type is set in the url, validate it, and use it in the query.
-    if (!empty($_GET['graph'])) {
-      $def = $rdf_storage->getGraphDefinitions();
-      if (is_string($_GET['graph']) && isset($def[$_GET['graph']])) {
+    $graph = $request->get('graph');
+    if (!empty($graph)) {
+      $definitions = $rdf_storage->getGraphDefinitions();
+      if (isset($definitions[$graph])) {
         // Use the graph to build the list.
-        $query->setGraphType([$_GET['graph']]);
+        $query->setGraphType([$graph]);
       }
     }
     else {
       $query->setGraphType($rdf_storage->getGraphHandler()->getEntityTypeEnabledGraphs());
     }
 
+    if ($rid = $request->get('rid') ?: NULL) {
+      $rid = in_array($rid, array_keys($bundle_info->getBundleInfo('rdf_entity'))) ? [$rid] : NULL;
+    }
+
+    $query->condition('rid', $rid, 'IN');
+    // Special treatment for 'solution' and 'asset_release'.
+    // @see https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-3126
+    if ($rid[0] === 'asset_release') {
```

Comment:
Yes, no problem. But that I added because very often I have to check if an entity was migrated. It's very practical to have that filter in place. I will find a workaround.